herculens/MassModel/mass_model.py

```diff
@@ -52,12 +79,12 @@ def potential(self, x, y, kwargs, k=None):
         :param k: only evaluate the k-th lens model
         :return: lensing potential in units of arcsec^2
         """
-        x = np.array(x, dtype=float)
-        y = np.array(y, dtype=float)
+        # x = np.array(x, dtype=float)
+        # y = np.array(y, dtype=float)
         if isinstance(k, int):
             return self.func_list[k].function(x, y, **kwargs[k])
         bool_list = self._bool_list(k)
-        potential = np.zeros_like(x)
```

Comment:
this is a code comment: if x is an array the return object is not in the same format/dimension